src/bootstrap/bootstrap.py

```diff
@@ -315,7 +315,7 @@ def build_triple(self):
         try:
             ostype = subprocess.check_output(['uname', '-s']).strip().decode(default_encoding)
             cputype = subprocess.check_output(['uname', '-m']).strip().decode(default_encoding)
-        except (subprocess.CalledProcessError, WindowsError):
+        except subprocess.CalledProcessError:
```

Comment:
Yes, as I suspected if the execution of 'isainfo' fails then Python will trip over the Exception handler at runtime:

```
Traceback (most recent call last):
  File "./x.py", line 20, in <module>
    bootstrap.main()
  File "/builds/srwalker/rsdev/rust.git/src/bootstrap/bootstrap.py", line 492, in main
    rb.build = rb.build_triple()
  File "/builds/srwalker/rsdev/rust.git/src/bootstrap/bootstrap.py", line 358, in build_triple
    except (subprocess.CalledProcessError, WindowsError):
NameError: global name 'WindowsError' is not defined
```

...and annoyingly, despite pydoc's claims otherwise, Python doesn't raise CalledProcessError here; it raises an OSError() (if the command is missing for some reason).  My guess is that it will only raise CalledProcessError() if execution of the command returns failure, as opposed to being unable to execute the process at all.  So this should actually catch both CalledProcessError and OSError.  I will apply that fix and your other requested change and try the full build again.